invenio_rdm_records/records/pids/providers/datacite.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite DOI Provider."""
+
+from datacite import DataCiteRESTClient
+from datacite.errors import DataCiteError, HttpError
+from invenio_pidstore.models import PIDStatus
+
+from .base import BaseProvider
+
+
+class DataCiteProvider(BaseProvider):
+    """Base Provider class.
+
+    Note that DataCite is only contacted when a DOI is reserved or
+    registered, or any action posterior to it. Its creation happens
+    only at PIDStore level.
+    """
+
+    def __init__(self, name, client=None, pid_type="doi",
```

Comment:
I think there might a misunderstanding: for me, the client param is basically one a account of a given provider, e.g. zenodo or CDS on DataCite.
Then, in the `datacite` Python module, client is instead the REST API Python client lib.

What about if make `client` mandatory and not all the rest?
```
def __init__(self, name, client, pid_type="doi", default_status=PIDStatus.NEW, **kwargs):
    self.api_client = DataCiteRESTClient(client.username, client.password, client.prefix, client.test_mode)
```

Reply:
I see... thanks! changed :)